src/main/java/ca/uhn/fhir/jpa/demo/FhirServerConfig.java

```diff
@@ -62,41 +64,47 @@ public DatabaseBackedPagingProvider databaseBackedPagingProvider() {
 	 * A URL to a remote database could also be placed here, along with login credentials and other properties supported by BasicDataSource.
 	 */
 	@Bean(destroyMethod = "close")
-	public DataSource dataSource() {
+	public DataSource dataSource() throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
 		BasicDataSource retVal = new BasicDataSource();
-		retVal.setDriver(new org.apache.derby.jdbc.EmbeddedDriver());
-		retVal.setUrl("jdbc:derby:directory:target/jpaserver_derby_files;create=true");
-		retVal.setUsername("");
-		retVal.setPassword("");
+		Driver driver = (Driver) Class.forName(HapiProperties.getDataSourceDriver()).getConstructor().newInstance();
+		retVal.setDriver(driver);
+		retVal.setUrl(HapiProperties.getDataSourceUrl());
+		retVal.setUsername(HapiProperties.getDataSourceUsername());
+		retVal.setPassword(HapiProperties.getDataSourcePassword());
 		return retVal;
 	}
 
 	@Override
 	@Bean()
 	public LocalContainerEntityManagerFactoryBean entityManagerFactory() {
 		LocalContainerEntityManagerFactoryBean retVal = super.entityManagerFactory();
-		retVal.setPersistenceUnitName("HAPI_PU");
-		retVal.setDataSource(dataSource());
+		retVal.setPersistenceUnitName(HapiProperties.getPersistenceUnitName());
+
+		try {
+			retVal.setDataSource(dataSource());
```

Comment:
It'd probably make more sense to just re-throw the exceptions below, given that we won't be able to start without this succeeding